### PR TITLE
add soap connection to cc service and connection tests

### PR DESCRIPTION
**Pull request by eeren**

### CODE REVIEW

File: carrental.h2.trace.db [hunk 1]
```diff
@@ -0,0 +1,4 @@
+2020-04-12 20:44:20 jdbc[13]: exception
+org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax Fehler in SQL Befehl "SELECT * FROM CAR RENTAL CAR[*]"
+Syntax error in SQL statement "SELECT * FROM CAR RENTAL CAR[*]"; SQL statement:
+SELECT * FROM CAR RENTAL CAR [42000-200]
```
Comment (JamHaven) [inline carrental.h2.trace.db:4]: What is this file?

File: pom.xml [hunk 2]
```diff
@@ -91,6 +111,27 @@
...
+							<goal>generate</goal>
+						</goals>
+					</execution>
+				</executions>
+				<configuration>
+					<schemaLanguage>WSDL</schemaLanguage>
+					<generatePackage>com.consumingwebservice.wsdl</generatePackage>
+					<schemas>
+						<schema>
+							<url>http://localhost:50923/Service1.svc?wsdl</url>
```
Comment (JamHaven) [inline pom.xml]: Only the valid URLs will be merged into master. in this case: https://currencyconverter.azurewebsites.net/Service.svc?wsdl

File: src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		getWebServiceTemplate().setMarshaller(new SoapMarshaller().marshaller());
+		getWebServiceTemplate().setUnmarshaller(new SoapMarshaller().marshaller());		
+	}
+	
+	public GetCurrencyCodesResponse getCurrencyCodesResponse() {
+		try {
+			GetCurrencyCodes request = new GetCurrencyCodes();
+			setupMarshaller();
+			GetCurrencyCodesResponse response = (GetCurrencyCodesResponse) getWebServiceTemplate()
+		        .marshalSendAndReceive("http://localhost:50923/Service1.svc/soap", request,
```
Comment (JamHaven) [inline src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java]: Pleas change the localhost URL

File: src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		            new SoapActionCallback("http://tempuri.org/IService1/GetCurrencyCodes"));
```
Comment (JamHaven) [inline src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java]: Tempuri.org ? What is this used for?  the url leads to 404

File: src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+	public ConvertCurrencyResponse convertCurrency() {
+		try {
+			ConvertCurrency request = new ConvertCurrency();
+			setupMarshaller();
+			ObjectFactory factory = new ObjectFactory();
+			request.setValue(factory.createConvertCurrencyValue("1"));
+			request.setToCurrency(factory.createConvertCurrencyToCurrency("JPY"));
+			ConvertCurrencyResponse response = (ConvertCurrencyResponse)getWebServiceTemplate()
+					 .marshalSendAndReceive("http://localhost:50923/Service1.svc/soap", request,
+							  new SoapActionCallback("http://tempuri.org/IService1/ConvertCurrency"));
```
Comment (JamHaven) [inline src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java]: Same here, see comments above

File: src/main/java/pacApp/pacSoapConnector/context/SoapMarshaller.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+	  public Jaxb2Marshaller marshaller() {
+	    Jaxb2Marshaller marshaller = new Jaxb2Marshaller();
+	    marshaller.setContextPath("com.consumingwebservice.wsdl");
+	    return marshaller;
+	  }
+
+	  @Bean
+	  public SoapConvertCurrencyConnector cClient(Jaxb2Marshaller marshaller) {
+		SoapConvertCurrencyConnector client = new SoapConvertCurrencyConnector();
+	    client.setDefaultUri("http://localhost:50923/Service1.svc?wsdl");
```
Comment (JamHaven) [inline src/main/java/pacApp/pacSoapConnector/context/SoapMarshaller.java]: Please change to Production URL

File: src/main/java/pacApp/pacSoapConnector/context/TestService.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+import pacApp.pacSoapConnector.SoapConvertCurrencyConnector;
+
+@Service
+public class TestService implements ISoapTestService {
+	
+	@Override
+	public String runTest() {
+		new SoapConvertCurrencyConnector().getCurrencyCodesResponse();
+		return null;
```
Comment (JamHaven) [inline src/main/java/pacApp/pacSoapConnector/context/TestService.java]: return null? I guess this test is no finished?

File: src/main/java/pacApp/pacSoapConnector/context/TestService2.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+import pacApp.pacSoapConnector.SoapConvertCurrencyConnector;
+
+@Service
+public class TestService2 implements ISoapTestService{
+
+	@Override
+	public String runTest() {
+		new SoapConvertCurrencyConnector().convertCurrency();		
+		return null;
```
Comment (JamHaven) [inline src/main/java/pacApp/pacSoapConnector/context/TestService2.java]: return null? I guess this test is no finished?

**Review by JamHaven — CHANGES_REQUESTED** [review]
Read the comments.
Replace the localhost URLs with the one from production

File: src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java (hunk 3 above)
Comment (eeren) [reply src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java]: i will change the urls, if the Soap server run on cloud. 
The new version of Soap is on Head Rep.

File: pom.xml (hunk 2 above)
Comment (eeren) [reply pom.xml]: i will change the urls, if the Soap server run on cloud. 
The new version of Soap is on Head Rep.

File: carrental.h2.trace.db (hunk 1 above)
Comment (eeren) [reply carrental.h2.trace.db:4]: can be ignored

File: src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java (hunk 4 above)
Comment (eeren) [reply src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java]: Yes, for this class required!

Comment (eeren) [reply src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java]: This url is definded in the Soap WSDL

File: src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java (hunk 5 above)
Comment (eeren) [reply src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java]: This url is definded in the Soap WSDL

File: src/main/java/pacApp/pacSoapConnector/context/SoapMarshaller.java (hunk 6 above)
Comment (eeren) [reply src/main/java/pacApp/pacSoapConnector/context/SoapMarshaller.java]: see above! I will change all URLs if the SOAP Server Version run on cloud

File: src/main/java/pacApp/pacSoapConnector/context/TestService.java (hunk 7 above)
Comment (eeren) [reply src/main/java/pacApp/pacSoapConnector/context/TestService.java]: Is a test class. The class is now deleted.

File: src/main/java/pacApp/pacSoapConnector/context/TestService2.java (hunk 8 above)
Comment (eeren) [reply src/main/java/pacApp/pacSoapConnector/context/TestService2.java]: Is a test class. The class is now deleted.

File: src/main/java/pacApp/pacSoapConnector/context/SoapMarshaller.java (hunk 6 above)
Comment (JamHaven) [reply src/main/java/pacApp/pacSoapConnector/context/SoapMarshaller.java]: They are live since I wrote this. Please change

Comment (eeren) [reply src/main/java/pacApp/pacSoapConnector/context/SoapMarshaller.java]: The Wsdl has changed.

Comment (eeren) [reply src/main/java/pacApp/pacSoapConnector/context/SoapMarshaller.java]: I add the soap auth functionality.

Comment (JamHaven) [reply src/main/java/pacApp/pacSoapConnector/context/SoapMarshaller.java]: The WSDL will be taken from the "production" currency converter deployment (currently dev branch). The service has auto-deploy for this branch. If you changed something, which there should have been a merge request for, than it already auto deployed a new version to the cloud --> you can change it from localhost.

File: pom.xml (hunk 2 above)
Comment (eeren) [reply pom.xml]: done

File: src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java (hunk 3 above)
Comment (eeren) [reply src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java]: done

**Comment by philsch91** [comment]
@JamHaven @JamHaven are these additions ready and working for a merge?

**Pull request merged**